src/main/java/reactor/core/publisher/Mono.java

```diff
@@ -2732,6 +2733,82 @@ public final Disposable subscribe(Consumer<? super T> consumer,
 	}
 
 	/**
+	 * Subscribe to a returned fallback publisher when any error occurs.
+	 *
+	 * <p>
+	 * <img class="marble" src="https://raw.githubusercontent.com/reactor/reactor-core/v3.0.6.RELEASE/src/docs/marble/otherwise.png" alt="">
+	 * <p>
+	 * @param fallback the function to map an alternative {@link Mono}
+	 *
+	 * @return an alternating {@link Mono} on source onError
+	 *
+	 * @see Flux#switchOnError
+	 */
+	public final Mono<T> switchOnError(Function<? super Throwable, ? extends Mono<? extends
+			T>> fallback) {
+		return onAssembly(new MonoSwitchOnError<>(this, fallback));
+	}
+
+	/**
+	 * Subscribe to a returned fallback publisher when an error matching the given type
+	 * occurs.
+	 * <p>
+	 * <img class="marble" src="https://raw.githubusercontent.com/reactor/reactor-core/v3.0.6.RELEASE/src/docs/marble/otherwise.png"
+	 * alt="">
+	 *
+	 * @param type the error type to match
+	 * @param fallback the {@link Function} mapping the error to a new {@link Mono}
+	 * sequence
+	 * @param <E> the error type
+	 *
+	 * @return a new {@link Mono}
+	 * @see Flux#switchOnError
+	 */
+	public final <E extends Throwable> Mono<T> switchOnError(Class<E> type,
+			Function<? super E, ? extends Mono<? extends T>> fallback) {
+		Objects.requireNonNull(type, "type");
+		@SuppressWarnings("unchecked")
+		Function<? super Throwable, Mono<? extends T>> handler = (Function<? super
+				Throwable, Mono<? extends T>>)fallback;
+		return switchOnError(type::isInstance, handler);
+	}
+
+	/**
+	 * Subscribe to a returned fallback publisher when an error matching the given type
```

Comment:
the javadoc was inconsistent here: it's "when an error matching the given predicate", not "type"